addons/weather/functions/fnc_getMapData.sqf

```diff
@@ -35,23 +39,42 @@ GVAR(WindDirectionProbabilities) = [
     [0.06, 0.37, 0.05, 0.03, 0.18, 0.04, 0.02, 0.02]  // December
 ];
 
-// Check if the wind data is defined in the map config
-if (isArray (configFile >> "CfgWorlds" >> worldName >> "ACE_WindSpeedMean")) then {
-    GVAR(WindSpeedMin)               = getArray (configFile >> "CfgWorlds" >> worldName >> "ACE_WindSpeedMin");
-    GVAR(WindSpeedMean)              = getArray (configFile >> "CfgWorlds" >> worldName >> "ACE_WindSpeedMean");
-    GVAR(WindSpeedMax)               = getArray (configFile >> "CfgWorlds" >> worldName >> "ACE_WindSpeedMax");
-    GVAR(WindDirectionProbabilities) = getArray (configFile >> "CfgWorlds" >> worldName >> "ACE_WindDirectionProbabilities");
-};
+GVAR(TempDay) = [1, 3, 9, 14, 19, 23, 25, 24, 21, 13, 7, 2];
+GVAR(TempNight) = [-4, -3, 0, 4, 9, 12, 14, 14, 10, 6, 2, -2];
+GVAR(Humidity) = [82, 80, 78, 70, 71, 72, 70, 73, 78, 80, 83, 82];
+
+GVAR(currentTemperature) = 20;
+GVAR(currentHumidity) = 0.5;
 
-// Check if the weather data is defined in the map config
-if (isArray (configFile >> "CfgWorlds" >> worldName >> "ACE_TempDay")) exitWith {
-    GVAR(TempDay)   = getArray (configFile >> "CfgWorlds" >> worldName >> "ACE_TempDay");
-    GVAR(TempNight) = getArray (configFile >> "CfgWorlds" >> worldName >> "ACE_TempNight");
-    GVAR(Humidity)  = getArray (configFile >> "CfgWorlds" >> worldName >> "ACE_Humidity");
+// Get all non inherited arrays to filter maps that inherit from Stratis/Altis/Tanoa
+private _nonInheritedArrays = configProperties [configFile >> "CfgWorlds" >> _worldName, "isArray _x", false];
+// And check if any custom non-inherited weather is defined through config and use that if so
+if ((configFile >> "CfgWorlds" >> _worldName >> "ACE_TempDay") in _nonInheritedArrays) exitWith {
+    if (isArray (configFile >> "CfgWorlds" >> _worldName >> "ACE_TempDay")) then {
```

Comment:
If one weather value is defined I think it's safe to assume all of them are. I don't think these checks are necessary.